compiler/quilt/nodes.py

```diff
@@ -139,6 +140,38 @@ def _data(self):
 
         return store.load_dataframe(hash_list)
 
+def _create_filter_func(filter_dict):
+    filter_name = filter_dict.pop('name', None)
+    if filter_name is not None and not isinstance(filter_name, string_types):
+        raise ValueError("Invalid 'name'")
+
+    filter_meta = filter_dict.pop('meta', None)
+    if filter_meta is not None and not isinstance(filter_meta, dict):
+        raise ValueError("Invalid 'meta'")
+
+    if filter_dict:
+        raise ValueError("Unexpected data in the filter: %r" % filter_dict)
```

Comment:
I did `foo._filter({"color":"orange"})` and this message confused me.
1. we probably shouldn't require the "meta" key on top? (i know, the spec shows it.) we just make it clear in the filter documentation/contract that it filters meta only.
1. the error message should be something like this: `Unexpected key {} passed to _filter: {}` And maybe we won't need that message anymore?

Reply:
If we don't require "meta", how do we handle the node name? (It can't be a part of the metadata -  nodes can't know their own names. If we add a hack just for the filter, then it becomes a yet another reserved word...)

Reply:
oh i see. i had forgotten about that. i thought the node name was a second positional argument. would it work in that case?

btw - i know understand what `.pop('name',...) does above.